hil-test/tests/spi_slave.rs

```diff
@@ -31,19 +34,32 @@ struct Context {
 }
 
 struct BitbangSpi {
-    sclk: Output<'static>,
-    mosi: Output<'static>,
+    sclk: OutputSignal,
+    mosi: OutputSignal,
     miso: InputSignal,
-    cs: Output<'static>,
+    cs: OutputSignal,
 }
 
 impl BitbangSpi {
     fn new(
-        sclk: Output<'static>,
-        mosi: Output<'static>,
-        miso: InputSignal,
-        cs: Output<'static>,
+        mut sclk: OutputSignal,
+        mut mosi: OutputSignal,
+        mut miso: InputSignal,
+        mut cs: OutputSignal,
     ) -> Self {
+        // TODO remove this (#2273)
+        // FIXME: devise a public API for signals
+        miso.enable_input(true, unsafe { esp_hal::Internal::conjure() });
```

Comment:
I hate what I had to do to this poor test, hopefully this will force me to clean this up :)

Reply:
Will fix after #2427